org.eclipse.scanning.malcolm.core/src/org/eclipse/scanning/malcolm/core/AbstractMalcolmDevice.java

```diff
@@ -215,10 +215,12 @@ protected void subscribeToConnectionStateChange(IMalcolmListener<Boolean> listen
 	 * @throws InterruptedException
 	 */
 	protected MalcolmMessage send(MalcolmMessage message, long timeout) throws MalcolmDeviceException, InterruptedException, ExecutionException, TimeoutException {
+		logger.debug("Sending message to malcolm device: {}", message);
 	    return asynch(()->connector.send(this, message), timeout);
 	}
 
 	protected MalcolmMessage call(MalcolmMethod method, long timeout, DeviceState... states) throws MalcolmDeviceException, InterruptedException, ExecutionException, TimeoutException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.eclipse.scanning.api.malcolm.MalcolmDeviceException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)